lib/src/main/java/uk/gov/di/ipv/cri/kbv/api/security/SoapTokenRetriever.java

```diff
@@ -34,7 +43,49 @@ public String getSoapToken() {
             }
             retry++;
         }
-        return token;
+
+        if (!validTokenFetch && !isCachedTokenValid()) {
+            try {
+                if (token != null
+                        && SoapTokenUtils.isTokenValid(SoapTokenUtils.decodeTokenPayload(token))) {
+                    cachedToken = token;
+                    LOGGER.info(
+                            "Updated cached token with the one received from Experian. The token given by Experian is valid but not within our threshold, using anyway...");
+                }
+            } catch (Exception ignore) {
+                LOGGER.warn("Cached SOAP token and token from Experian are both invalid");
```

Comment:
can we have visibility of what went wrong so 

`LOGGER.warn("Cached SOAP token and token from Experian are both invalid", e.getMessage());`

Reply:
It'll be a Jackson error. The errors are expectedly thrown errors. I don't want to accidentally expose anything via logging or get into the mess of redaction in the future if Experian some how decide to spit out sensitive information in the future.